applications/dashboard/models/class.usermodel.php

```diff
@@ -2212,8 +2212,8 @@ public function saveRoles($UserID, $RoleIDs, $RecordEvent) {
         $OldRoleIDs = array();
         $OldRoleData = $this->SQL
             ->select('ur.RoleID, r.Name')
-            ->from('Role r')
-            ->join('UserRole ur', 'r.RoleID = ur.RoleID')
+            ->from('UserRole ur')
+            ->join('Role r', 'r.RoleID = ur.RoleID', 'left')
```

Comment:
This is the main fix. Always bring in all entries from GDN_UserRole regardless of them existing in GDN_Role.